api/handlers.go

```diff
@@ -185,6 +185,33 @@ func (a *API) List(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type mountState struct {
+	w          http.ResponseWriter
+	err        error
+	ut         *config.UseMount
+	driver     storage.MountDriver
+	driverOpts storage.DriverOptions
+	volConfig  *config.Volume
+}
+
+// triggered on any failure during call into mount.
+func (a *API) clearMount(ms mountState) {
+	log.Errorf("MOUNT FAILURE: %v", ms.err)
+
+	if err := ms.driver.Unmount(ms.driverOpts); err != nil {
+		// literally can't do anything about this situation. Log.
+		log.Errorf("Failure during unmount after failed mount: %v %v", err, ms.err)
+	}
+
+	if err := a.Lock.ClearLock(ms.ut, (*a.Global).Timeout); err != nil {
```

Comment:
I guess here you can call `clearLock(ut config.UseLocker)`, this will eliminate `volConfig` from the struct. And, you can return `err` and wrap it with `HTTPError` in the caller `Mount`(this will remove w, err from the struct). Hope it will make this function a little cleaner.